src/containers/admin/GradingPage/index.js

```diff
@@ -1,63 +1,114 @@
-    });
+    this.changeGrade = this.changeGrade.bind(this);
+    this.nextTeam = this.nextTeam.bind(this);
   }
 
-  componentDidUpdate(prevProps) {
-    if (prevProps.match.params.team === this.props.match.params.team) {
-      return;
-    }
+  componentDidMount() {
+    // give time for clients' answers to get submitted
+    this.timeout = setTimeout(() => {
+      // get the round, questions, and teams to grade
+      this.firebase.getGame((game) => {
+        // get the round from stage, ex. round1-grading => round1
+        const round = game.stage.split('-')[0];
+        this.setState({
+          // filter out no bonus
+          questions: game[round].filter((q) => (q.q !== '')),
+          teams: game.teams,
+          teamNames: Object.keys(game.teams),
+          currentTeamNum: 0,
+          // 11 false in case of bonus, only will display ten if no, and last false will not matter
+          teamCorrects: [false, false, false, false, false, false, false, false, false, false, false],
+          round,
+        });
+      });
+    }, 2000);
+  }
+
+  // just saving this because it might be better than that derived state thing on question page
+  // componentDidUpdate(prevProps) {
+  //   if (prevProps.match.params.team === this.props.match.params.team) {
+  //     return;
+  //   }
+
+  //   this.setState({
+  //     corrects: [false, false, false, false, false, false, false, false, false, false]
+  //   });
+  // }
 
-    this.setState({
-      correctAnswers: [false, false, false, false, false, false, false, false, false, false]
-    });
+  componentWillUnmount() {
+    // have to clear every timeout we use just in case it doesn't finish
+    clearTimeout(this.timeout);
   }
 
+  // either true or false in val
   changeGrade(i, val) {
-    const corrects = [...this.state.corrects];
-    corrects[i] = val;
-    this.setState({ corrects });
+    const teamCorrects = [...this.state.teamCorrects];
+    teamCorrects[i] = val;
+    this.setState({ teamCorrects });
   }
 
-  render() {
-    if (!this.props.match.params.team) {
-      return <Redirect to="/admin/login" />;
+  nextTeam() {
+    // add one point for each correct answer
+    this.teamScores.push(this.state.teamCorrects.filter(Boolean).length);
+
+    const currentTeamNum = this.state.currentTeamNum + 1;
+    // keep repeating if we have teams left
+    if (currentTeamNum < this.state.teamNames.length) {
+      this.setState({
+        teamCorrects: [false, false, false, false, false, false, false, false, false, false, false],
+        currentTeamNum,
+      }, () => window.scrollTo(0, 0));
+    } else {
+      // otherwise save the scores
+      const teams = JSON.parse(JSON.stringify(this.state.teams));
+      for (let i = 0; i < this.state.teamNames.length; i++) {
+        const name = this.state.teamNames[i];
+        // add to the current score if possible
+        let score = teams[name].score || 0;
+        score += this.teamScores[i];
```

Comment:
nit: rename `teamScores` to `currentRoundTeamScores`, so that it's less ambiguous